services/app-api/utils/other/other.ts

```diff
@@ -102,3 +103,63 @@ export const getLastCreatedWorkPlan = async (
   // If there wasn't an eligble work plan to copy from, return undefined
   return { workPlanMetadata: undefined, workPlanFieldData: undefined };
 };
+
+export const getReportYear = (
+  reportType: string,
+  //reportData: unvalidatedMetadata || workPlanMetadata
+  reportData: any,
+  isCopyOver: boolean = false
+): number => {
+  if (isCopyOver) {
+    if (typeof reportData?.copyReport?.reportYear !== "number") {
+      throw new Error("Invalid value for reportYear");
+    }
+    const prevReportYear = reportData?.copyReport?.reportYear;
+    const prevReportPeriod = reportData?.copyReport?.reportPeriod;
+
+    return prevReportPeriod == 2 ? prevReportYear + 1 : prevReportYear;
```

Comment:
do we want strict equals `===` here after our concerns about type safety? (though I guess the number check above should take care of any other case)

Reply:
Oops this was a copy/paste...you are right